File: back/app/models/Ingredient.scala
package models

import drools.recommendation.IngredientType
import scalikejdbc._

case class Ingredient(
    id: Long,
    name: String,
    fats: Double,
    carbs: Double,
    protein: Double,
    `type`: IngredientType
)

object Ingredient extends SQLSyntaxSupport[Ingredient] {

  override def tableName: String = "ingredients"

  def apply(rs: WrappedResultSet, rn: ResultName[Ingredient]): Ingredient =
    Ingredient(
      id = rs.get[Long](rn.id),
      name = rs.get[String](rn.name),
      fats = rs.get[Double](rn.fats),
      carbs = rs.get[Double](rn.carbs),
      protein = rs.get[Double](rn.protein),
      `type` = IngredientType.valueOf(rs.get[String](rn.`type`))
    )

}

File: back/app/models/Ingredient.java
package models;

import drools.recommendation.IngredientType;

import javax.persistence.*;

@Entity
@Table(name = "ingredients")
public class Ingredient {

    @Id
    @Column
    Long id;

    @Column
    String ingredient;

    @Column
    Double fats;

    @Column
    Double protein;

    @Column
    Double carbs;

    @Column
    @Enumerated(EnumType.STRING)
    IngredientType type;

    public Ingredient(Long id, String ingredient, Double fats, Double protein, Double carbs, IngredientType type) {
        this.id = id;
        this.ingredient = ingredient;
        this.fats = fats;
        this.protein = protein;
        this.carbs = carbs;
        this.type = type;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getIngredient() {
        return ingredient;
    }

    public void setIngredient(String ingredient) {
        this.ingredient = ingredient;
    }

    public Double getFats() {
        return fats;
    }

    public void setFats(Double fats) {
        this.fats = fats;
    }

    public Double getProtein() {
        return protein;
    }

    public void setProtein(Double protein) {
        this.protein = protein;
    }

    public Double getCarbs() {
        return carbs;
    }

    public void setCarbs(Double carbs) {
        this.carbs = carbs;
    }

    public IngredientType getType() {
        return type;
    }

    public void setType(IngredientType type) {
        this.type = type;
    }

}
